add fonts to matplotlib

jan-janssen opened the pull request:
@conda-forge/staged-recipes 

Checklist

- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages
- [ ] Build number is 0
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there



Motivation 

On Linux common fonts are not available dude to license restrictions from Microsoft.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/cabextract```, ```recipes/matplotlib-corefonts```) and found some lint.

Here's what I've got...


For **recipes/matplotlib-corefonts**:

 * about/license_family 'EULA' not allowed. Allowed families are AGPL,
APACHE, BSD, CC, GPL, GPL2, GPL3, LGPL, MIT, NONE, OTHER, PROPRIETARY,
PSF, and PUBLIC-DOMAIN.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/cabextract```, ```recipes/matplotlib-corefonts```) and found it was in an excellent condition.

ocefpaf commented on recipes/cabextract/build.sh:
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+./configure
+make
```
Is there a make check we can run?

jan-janssen replied:
At least I am not aware of such an option.

ocefpaf commented on recipes/cabextract/meta.yaml:
```diff
@@ -0,0 +1,41 @@
+
+test:
+  commands:
+    - conda inspect linkages {{ name }}  # [not win]
```
There is no need to run inspect any more. That is done by default in all build by conda-build. Can you add a CLI test for the `cabextract`?

ocefpaf commented on recipes/cabextract/meta.yaml:
```diff
@@ -0,0 +1,41 @@
+
+about:
+  home: https://www.cabextract.org.uk
+  license: GPL3
```
Please use the SPDX short version: `GPL-3.0`.

ocefpaf commented on recipes/matplotlib-corefonts/LICENSE:
```diff
@@ -0,0 +1,53 @@
+1. GRANT OF LICENSE. This EULA grants you the following rights:
+
+    Installation and Use. You may install and use an unlimited number of copies of the SOFTWARE PRODUCT.
+    Reproduction and Distribution. You may reproduce and distribute an unlimited number of copies of the SOFTWARE PRODUCT; provided that each copy shall be a true and complete copy, including all copyright and trademark notices, and shall be accompanied by a copy of this EULA. Copies of the SOFTWARE PRODUCT may not be distributed for profit either on a standalone basis or included as part of your ...
```
Looks like we can re-distribute it but I am not a lawyer.

jan-janssen replied:
@ocefpaf Me neither, is there anybody in the conda-forge team who can help out with this or anybody with contact to microsoft.

scopatz replied:
This is technically fine for us to redistribute, and I think we should have it as a package.  However, given some of the restrictions in the license, I (personally) would not recomend that we have any packages that depend on it.  I think people should explictly have to include it in their environment.

ocefpaf commented on recipes/matplotlib-corefonts/meta.yaml:
```diff
@@ -0,0 +1,45 @@
+
+requirements:
+  build:
+    - cabextract
```
There will be a race condition in the CIs when submitting both recipes together.
As long as you are fine restart the `matplotlib-corefonts` CI later this is fine. Otherwise send a separate PRs.

scopatz commented on recipes/matplotlib-corefonts/meta.yaml:
```diff
@@ -0,0 +1,45 @@
+{% set name = "matplotlib-corefonts" %}
```
Actually, under [section 2](https://github.com/conda-forge/staged-recipes/pull/8497/files#diff-13de976cfe10a42b2d3a376d8d8d3b2dR23) of the license, it is not allowed to rename this project.  I think we must call this `corefonts` and we cannot rebrand this as part of matplotlib.

scopatz replied:
Or maybe even better, call it `mscorefonts`

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/cabextract```, ```recipes/mscorefonts```) and found it was in an excellent condition.

jan-janssen commented:
@ocefpaf and @scopatz - thank you very much for your feedback. I updated the recipes accordingly and fixed some minor bugs. From my perspective the recipe is ready to be merged.

scopatz commented:
Thanks @jan-janssen!
